### DISCUSSION TITLE
Using cache.invalidate TS is complaining about SystemFields.id?: string | number

### DISCUSSION THREAD

**Discussion by frederikhors** [urql-graphql/urql#1462] [Q&A] [answered]
I have this code:

```ts
type Updates = {
  Mutation: {
    playerCreate: UpdateResolver;
    playerDelete: UpdateResolver;
  };
};

const updates: Updates = {
  Mutation: {
    playerCreate: (result, args, cache): UpdateResolver => {
      // ...
    },

    playerDelete: (result, args, cache, info): UpdateResolver => {
      if (result.playerDelete) {
        cache.invalidate({ __typename: "Player", id: info.variables.id }); // here the problem!
      }
    },
  },
};
```

Typescript is complaining about:

```
(property) SystemFields.id?: string | number
Type 'Variables | Scalar | Scalar[] | NullArray<Variables>' is not assignable to type 'string | number'.
  Type 'boolean' is not assignable to type 'string | number' .ts(2322)
```

**Why?**

**Comment by kitten** [comment, marked as answer]
It's because the type is _extremely_ generic with every possible GraphQL scalar value (or sometimes even objects, depending on the method call), so you'll need to cast `id`, e.g. `id as string`. That's until we add type generation for Graphcache.

Also on a side note, you likely _don't_ want to use `info.variables` and only use it if there's no alternative. Instead it looks like `args.id` or sth may be available for `playerDelete` depending on your schema.
